### ISSUE TITLE
Mapping to DSeqSpans runs into IndexError for specific strategy and example

### ISSUE THREAD

**Issue by FlorianGottschalk** [j-frei/DSeqMap4NLP#1]
Code to replicate error:

```python
from dseqmap4nlp import SpacySequenceMapper, LabelLoader

do_what_works: bool = False

text = '≥pT2.'
anns = [[1, 4, 'Diagnosis_or_Pathology']]

print(f'{text=}')
print(f'{anns=}')

mapper = SpacySequenceMapper(text=text, nlp='de')

annotationset = LabelLoader.from_text_spans(text_entries=anns, mapper=mapper)

if do_what_works:
    # works:
    filtered_spans = annotationset.toDSeqSpans(strategy=['expand']).withoutOverlaps(
        strategy='prefer_longest', merge_same_classes=True,
    )
else:
# does not work:
    filtered_spans = annotationset.toDSeqSpans(strategy=['contract', 'expand']).withoutOverlaps(
        strategy='prefer_longest', merge_same_classes=True,
    )

tokens = mapper.getDSeq()
labels = filtered_spans.toFormattedSequence(schema='IOB2')

print(f'{tokens=}')
print(f'{labels=}')
```

Error:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[11], line 22
     17     filtered_spans = annotationset.toDSeqSpans(strategy=['expand']).withoutOverlaps(
     18         strategy='prefer_longest', merge_same_classes=True,
     19     )
     20 else:
     21 # does not work:
---> 22     filtered_spans = annotationset.toDSeqSpans(strategy=['contract', 'expand']).withoutOverlaps(
     23         strategy='prefer_longest', merge_same_classes=True,
     24     )
     26 tokens = mapper.getDSeq()
     27 labels = filtered_spans.toFormattedSequence(schema='IOB2')

File ~/venv/lib/python3.10/site-packages/dseqmap4nlp/annotationset.py:43, in AnnotationSet.toDSeqSpans(self, strategy, mapper)
     41 while next_strategies:
     42     next_strategy, *next_strategies = next_strategies
---> 43     new_span = mapper.mapTextSpanToDSeqSpan((e_start, e_stop), mode=next_strategy)
     44     if new_span is not None:
     45         dseq_entries.append((*new_span, e_lbl))

File ~/venv/lib/python3.10/site-packages/dseqmap4nlp/basemapper.py:127, in BaseMapper.mapTextSpanToDSeqSpan(self, text_span, mode)
    124         start += 1
    126 # contract to next right sequence item, if we are outside the dseq items
--> 127 if self.txt2seq[start] == -1:
    128     while start < n_text and self.txt2seq[start] == -1:
    129         start += 1

IndexError: list index out of range
```

**Comment by j-frei** [comment]
Should be fixed now by commit 7bf9399ab87852068ea8244676115232147981ca. 

Thanks for reporting!

**Comment by FlorianGottschalk** [comment]
Thank you!
